components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -203,63 +182,76 @@ public Response updateResource(String resourceId, ResourceDetailsDTO updatedReso
      * @param resourceId resourceId of the resource which need to get deleted
      * @return Response with the status of resource deletion
      */
-    @Override
-    public Response deleteResource(String resourceId, MessageContext messageContext) {
 
-        Response response = null;
+    @Override
+    public Response deleteResource(String resourceId, MessageContext context) {
 
         try {
-            if (isResourceId(resourceId)) {
-
-                try {
-                    if (ResourceUtils.getResourceService().deleteResource(resourceId)) {
-                        return Response.status(Response.Status.NO_CONTENT).build();
-                    }
-
-                } catch (UMAServiceException e) {
-                    handleErrorResponse(e, log);
-                    log.error("Invalid request. ", e);
-
-                } catch (Throwable throwable) {
-                    handleErrorResponse((UMAException) throwable, log);
-                    log.error("Internal server error occurred. ", throwable);
+            if (isValidToken(context) && isResourceIdValid(resourceId)) {
+                if (ResourceUtils.getResourceService().deleteResource(resourceId)) {
+                    return Response.status(Response.Status.NO_CONTENT).build();
```

Comment:
Just wondering.. are we sending out 204 on successful delete ?

Reply:
In UMA specification after successful delete it's given to display as 204 (NO_CONTENT).

Reply:
Interesting..Can you please point me to the place where it says to send out 204 on successful deletion ? Also don't we need to validate the existence of resource ?  Send out 404 when the it's a non existing resource ?

Reply:
![image](https://user-images.githubusercontent.com/27561022/38587979-3306c3fe-3d41-11e8-815f-0f1a556f8054.png)

Reply:
This was given in the specification. So should I change the specifications?

Reply:
This says either 200 or 204. Since spec specifies 204 as well this should be fine. This is for a successful response. What if the resource doesn't exist ?